core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -771,12 +774,15 @@ public static Builder buildFrom(TableMetadata base) {
     private final Map<String, String> properties;
     private long currentSnapshotId;
     private List<Snapshot> snapshots;
-    private Map<String, SnapshotRef> refs;
+    private final Map<String, SnapshotRef> refs;
 
     // change tracking
     private final List<MetadataUpdate> changes;
     private final int startingChangeCount;
     private boolean discardChanges = false;
+    private Integer lastAddedSchemaId = null;
+    private Integer lastAddedSpecId = null;
+    private Integer lastAddedOrderId = null;
```

Comment:
nit: none of the other private varibles are explicitly set in definition.  Do we really need to set these to `null`?

Reply:
We explicitly set variables that are not `final`. Using `final` guarantees that they are set in a constructor, but there is no guarantee for mutable fields.